enterprise/server/remote_execution/platform/platform.go

```diff
@@ -259,7 +259,11 @@ func (p *ExecutorProperties) SupportsIsolation(c ContainerType) bool {
 // executor-specific overrides applied via the ApplyOverrides function.
 func ParseProperties(task *repb.ExecutionTask) (*Properties, error) {
 	m := map[string]string{}
-	for _, prop := range task.GetCommand().GetPlatform().GetProperties() {
+	p := task.GetAction().GetPlatform()
+	if p == nil || len(p.GetProperties()) == 0 {
+		p = task.GetCommand().GetPlatform()
+	}
```

Comment:
nit: can you consolidate this fallback logic into a helper func in `platform.go` like `platform.GetProto(action, command)`